File: python/src/BinarySearchList.py
import math


def binary_fn(haystack, needle):
    lo = 0
    hi = len(haystack)

    while lo < hi:
        mid = math.floor(lo + ((hi - lo) / 2))
        if haystack[mid] == needle:
            return True
        elif haystack[mid] < needle:
            lo = mid + 1
        else:
            hi = mid

    return False

File: src/day1/BinarySearchList.ts
export default function bs_list(haystack: number[], needle: number): boolean {
	let high = haystack.length;
	let low = 0;

	while (low < high) {
		const mid = Math.floor(low + (high - low) / 2);
		const val = haystack[mid];
		if (val === needle) {
			return true;
		}

		if (val > needle) {
			high = mid;
		} else {
			low = mid + 1;
		}
	}
	return false;
}
